components/support/nimbus-cli/src/output/server.rs

```diff
@@ -305,14 +348,67 @@ mod tests {
         let platform = "android";
         let deeplink = "fenix-dev-test://open-now";
 
-        let payload = StartAppPostPayload::new(platform, deeplink);
+        let payload = StartAppPostPayload::new(platform, deeplink, None);
         let _ = post_payload(&payload, &format!("127.0.0.1:{port}")).await?;
 
         // Check the index.html page
-        let s = get(port, "").await?;
+        let s = get(port, "/").await?;
         assert!(s.contains(deeplink));
 
         let _ = tx.send(());
         Ok(())
     }
+
+    #[tokio::test]
+    async fn test_posting_value_to_fake_remote_settings() -> Result<()> {
+        let port = 1237;
+        let (_, tx) = start_test_server(port)?;
+
+        let platform = "android";
+        let deeplink = "fenix-dev-test://open-now";
+        let value = json!({
+            "int": 1,
+            "boolean": true,
+            "object": {},
+            "array": [],
+            "null": null,
+        });
+        let payload = StartAppPostPayload::new(platform, deeplink, Some(&value));
+        let _ = post_payload(&payload, &format!("127.0.0.1:{port}")).await?;
+
+        // Check the fake Remote Settings page
+        let s = get(port, "/v1/buckets/BUCKET/collections/COLLECTION/records").await?;
+        assert_eq!(s, serde_json::to_string(&value)?);
+
+        let s = get(port, "/buckets/BUCKET/collections/COLLECTION/records").await?;
+        assert_eq!(s, serde_json::to_string(&value)?);
+
+        let _ = tx.send(());
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn test_getting_null_values_from_fake_remote_settings() -> Result<()> {
```

Comment:
Do we need a test for the `(StatusCode::NOT_MODIFIED, Json(Value::Null))` case too?